src/tools/rbd/action/Bench.cc

```diff
@@ -141,7 +141,10 @@ struct rbd_bencher {
   {
     if (io_type == IO_TYPE_WRITE || io_type == IO_TYPE_RW) {
       bufferptr bp(io_size);
-      memset(bp.c_str(), rand() & 0xff, io_size);
+      for (uint64_t i = 0; i < io_size; i++)
+      {
+        memset(bp.c_str() + i, 'a' + (rand() % 26), 1);
```

Comment:
What is the rationale for constraining to lower case ASCII characters?  The tool is writing bytes, so it seems weird to me.

Personally, I have never needed randomness here, but we did miss being able to specify a byte that gets written like you can do with other tools (e.g. `xfs_io`) on a few occasions.  How about making this behavior configurable with a new `--pattern-byte` option?

- For `--pattern-byte random`, do what is done in this PR, but without any constrains (i.e. use 0-255 range).
- Otherwise, for `--pattern-byte <pattern-byte>`, try to interpret `<pattern-byte>` as an unsigned integer in 0-255 range and fail the command with "rbd: invalid pattern byte" error if not successful.

Reply:
Thanks for reviewing - For my purposes restricting to lower case ASCII made it easier to differentiate data shards and parity shards at a glance but I agree it makes more sense to use the 0-255 for general purpose.

I have had a go at the above, adding the `--pattern-byte` option which takes either a single character or the string 'random', else fails with "rbd: the argument for option '--pattern-byte' is invalid". The default behaviour is random. 

Please let me know if there is more room for improvement.